src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -775,8 +779,9 @@ public EntryEditorPreferences getEntryEditorPreferences() {
                 getBoolean(DEFAULT_SHOW_SOURCE),
                 getBibtexKeyPatternPreferences(),
                 Globals.getKeyPrefs(),
-                getBoolean(AVOID_OVERWRITING_KEY));
-    }
+				getBoolean(AVOID_OVERWRITING_KEY),
+				getApiKeyPreferences());
```

Comment:
To make testing easier, it would be nice if you could directly pass the return value of `getWorldcatKey` (the string) to the fetcher.